agents/ppo/ppo.py

```diff
@@ -0,0 +1,52 @@
+from agents.ppo.types import *
+from typing import List
+import torch
+
+
+class PPOAgent(TorchAgent, config_name='ppo'):
+    def __init__(self, actor: BaseActor, critic: BaseCritic, discount_factor: float, ratio_clip: float):
+        super().__init__()
+        self._actor = actor
+        self._critic = critic
+        self._discount_factor = discount_factor
+        self._ratio_clip = ratio_clip
+
+    @classmethod
+    def create_from_config(cls, config):
+        return cls(
+            actor=BaseActor.create_from_config(config['actor']),
+            critic=BaseCritic.create_from_config(config['critic']),
+            discount_factor=config['discount_factor'],
+            ratio_clip=config['ratio_clip']
+        )
+
+    def forward(self, state: State) -> Tuple[State, Policy, Value, Value]:
+        next_state, policy = self._actor.act(state)
+        curr_v_func = self._critic.value(state)
+        q_estimate = self._critic.value(next_state)
+        return next_state, policy, curr_v_func, q_estimate
+
+    def loss(
```

Comment:
Let's separate `def loss` on different functions based their usage. From this code it is not true, that all this function is used only for loss computation. Some lines of code should be put into functions like  `def compute_advantage_value` for example